pkg/middleware/middleware.go

```diff
@@ -96,3 +96,63 @@ func (a *AuthMiddleware) Handler(next http.Handler) http.Handler {
 		next.ServeHTTP(w, r.WithContext(ctx))
 	})
 }
+
+func (a *AuthMiddleware) NonBlockingHandler(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		cookie, err := r.Cookie("auth")
+		if err != nil {
+			slog.Debug("Passing request to next handler without auth context")
+			ctx := context.WithValue(r.Context(), AuthContextKey, nil)
```

Comment:
This is not needed as to retrieve the user we do this:
```go
	user, ok := req.Context().Value(AuthContextKey)
```
That returns `nil` if the user is no value is associated with the key.